documentation/Generate-code-from-TypeSpec.md

```diff
@@ -0,0 +1,116 @@
+
+### Generate Code with code-gen-pipeline tool
+
+Install dependencies to use code-gen-pipeline,  
+```ps
+npm install -g @azure-tools/typespec-client-generator-cli
+npm install -g @microsoft/rush@5.92.0
+npm install -g @azure-tools/js-sdk-release-tools
+```
+
+Create a local json file named generatedInput.json with content similar to that shown below
+```
+   {
+    "dryRun": false,
+    "specFolder": <your-local-spec-repo-path>,
+    "headSha": <commit-id-you-want-to-generate-from>,
+    "repoHttpsUrl": "https://github.com/Azure/azure-rest-api-specs",
+    "relatedTypeSpecProjectFolder": [
+    "specification/SERVICE_DIRECTORY_NAME/PACKAGE_DIRECTORY_NAME/"
+  ]
+}
+```
+
+Run the command
+```
+code-gen-pipeline --inputJsonPath=<path-to-generatedInput.json> --outputJsonPath={path-to-anywhere(This will not be used locally, but this is required to use code-gen-pipeline)}  --typespecEmitter=@azure-tools/typespec-ts  --local 
+```
+
+This command will do:\
+1: generate code with typespec\
+2: generate CHANGELOG.md\
+3: update rush.json(if the generated package is new)\
+4: generate/update ci.mgmt.yml or ci.yml(if the generated package is new)
+
+
+
+### Generate Code with tsp-client tool
+
+Install `tsp-client` CLI tool
+
+```ps
+npm install -g @azure-tools/typespec-client-generator-cli
+```
+
+For initial set up, from the root of the SDK repo, call
+
+```
+tsp-client init -c <url-to-tspconfig>
+```
+
+For updating TypeSpec generated SDK, call below in the SDK module folder (`sdk/<service>/<module>`) where `tsp-location.yaml` exists
+
+```ps
+tsp-client update
+```
+
+If you want to generate SDK from your local tspconfig file from spec repo, you can run
+```
+tsp-client init --tsp-config=<your-local-tspconfig-path> --local-spec-repo=<your-local-service-path> --repo=<your-spec-repo-path> --commit=<your-target-commit>
+```
+here is a example command of this
+```
+tsp-client init --tsp-config=../azure-rest-api-specs/specification/connectedcache/ConnectedCache.Management/tspconfig.yaml --local-spec-repo=../azure-rest-api-specs/specification/connectedcache/ConnectedCache.Management --repo=../azure-rest-api-specs --commit=60c67f112c66537f1d006bf1b497857ccd2afacd
+```
+
+**Notice**
+If you use tsp-client to generate code and your generated SDK is new, you need to do two extra things:
+
+**1**: 
+You should add your new SDK in [rush.json](https://github.com/Azure/azure-sdk-for-js/blob/main/rush.json).
+Here is the example of the config
+```
+{
+  "packageName": <your-package-name>,
+  "projectFolder": <sdk/<service>/<module>>,
+  "versionPolicyName": <"management"-or-"client">
+}
+```
+
+**2**: You should add `ci.yml` or `ci.mgmt.yml` under `sdk/<service>/<module`. `ci.yml` is for `Data Plane SDKs` and `ci.mgmt.yml` is for `Mgmt Plane SDKs`. See [Create/Update the ci.yaml](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/steps-after-generations.md#createupdate-the-ciyaml)
+
+# Build
```

Comment:
it should be the `####` level section in `### Generate Code with tsp-client tool`, since the `### Generate Code with code-gen-pipeline tool` covers building package.